pkg/controllers/kube-controller-manager.go

```diff
@@ -16,89 +16,121 @@ limitations under the License.
+	configureErr error
 }
 
 func NewKubeControllerManager(cfg *config.MicroshiftConfig) *KubeControllerManager {
 	s := &KubeControllerManager{}
-	s.configure(cfg)
+	if err := s.configure(cfg); err != nil {
+		s.configureErr = err
+	}
 	return s
 }
 
 func (s *KubeControllerManager) Name() string           { return "kube-controller-manager" }
 func (s *KubeControllerManager) Dependencies() []string { return []string{"kube-apiserver"} }
 
-func (s *KubeControllerManager) configure(cfg *config.MicroshiftConfig) {
+func (s *KubeControllerManager) configure(cfg *config.MicroshiftConfig) error {
+	defaultConfigBytes, err := embedded.Asset(kcmDefaultConfigAsset)
+	if err != nil {
+		return fmt.Errorf("invalid asset: %q, error: %w", kcmDefaultConfigAsset, err)
+	}
+
 	certsDir := cryptomaterial.CertsDirectory(microshiftDataDir)
 	csrSignerDir := cryptomaterial.CSRSignerCertDir(certsDir)
-	kubeconfig := cfg.KubeConfigPath(config.KubeControllerManager)
-	kubeadmConfig := cfg.KubeConfigPath(config.KubeAdmin)
 
-	opts, err := kubecmoptions.NewKubeControllerManagerOptions()
-	if err != nil {
-		klog.Fatalf("%s initialization error command options: %v", s.Name(), err)
-	}
-	s.kubecmOptions = opts
-	s.kubeconfig = kubeconfig
+	s.kubeconfig = cfg.KubeConfigPath(config.KubeControllerManager)
+
+	kubeadmConfig := cfg.KubeConfigPath(config.KubeAdmin)
 	s.kubeadmConfig = kubeadmConfig
 
-	args := []string{
-		"--kubeconfig=" + kubeconfig,
-		"--service-account-private-key-file=" + microshiftDataDir + "/resources/kube-apiserver/secrets/service-account-key/service-account.key",
-		"--allocate-node-cidrs=true",
-		"--cluster-cidr=" + cfg.Cluster.ClusterCIDR,
-		"--authorization-kubeconfig=" + kubeconfig,
-		"--authentication-kubeconfig=" + kubeconfig,
-		"--root-ca-file=" + cryptomaterial.ServiceAccountTokenCABundlePath(certsDir),
-		"--bind-address=127.0.0.1",
-		"--secure-port=10257",
-		"--leader-elect=false",
-		"--use-service-account-credentials=true",
-		"--cluster-signing-cert-file=" + cryptomaterial.CACertPath(csrSignerDir),
-		"--cluster-signing-key-file=" + cryptomaterial.CAKeyPath(csrSignerDir),
+	overrides := kubecontrolplanev1.KubeControllerManagerConfig{
+		ExtendedArguments: map[string]kubecontrolplanev1.Arguments{
+			"kubeconfig": {s.kubeconfig},
+			"service-account-private-key-file": {
+				microshiftDataDir + "/resources/kube-apiserver/secrets/service-account-key/service-account.key",
+			},
+			"allocate-node-cidrs": {
+				"true",
+			},
+			"cluster-cidr": {
+				cfg.Cluster.ClusterCIDR,
+			},
+			"root-ca-file": {
+				cryptomaterial.ServiceAccountTokenCABundlePath(certsDir),
+			},
+			"bind-address": {
+				"127.0.0.1",
+			},
+			"secure-port": {
+				"10257",
+			},
+			"leader-elect": {
+				"false",
+			},
+			"use-service-account-credentials": {
+				"true",
+			},
+			"cluster-signing-cert-file": {
+				cryptomaterial.CACertPath(csrSignerDir),
+			},
+			"cluster-signing-key-file": {
+				cryptomaterial.CAKeyPath(csrSignerDir),
```

Comment:
seeing under this `kubelet-csr-signer-signer` - this should be used by both apiserver and kubelet right? Maybe just the naming is a it off?